hmpps-kotlin-spring-boot-test-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/test/kotlin/auth/WithMockAuthUser.kt

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.hmpps.test.kotlin.auth
+
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.context.SecurityContext
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.oauth2.jwt.Jwt
+import org.springframework.security.test.context.support.WithSecurityContext
+import org.springframework.security.test.context.support.WithSecurityContextFactory
+import org.springframework.util.Assert
+import uk.gov.justice.hmpps.kotlin.auth.AuthAwareAuthenticationToken
+import uk.gov.justice.hmpps.kotlin.auth.AuthSource
+import java.lang.annotation.Inherited
+
+/**
+ * Similar to {@link org.springframework.security.test.context.support.WithMockUser} but
+ * populates the security context with a {@link uk.gov.justice.hmpps.kotlin.auth.AuthAwareAuthenticationToken} instead.
```

Comment:
Could we maybe expand a little on when/why you would use the annotation in a test?